mne/preprocessing/tests/test_ica.py

```diff
@@ -918,6 +918,48 @@ def test_ica_ctf():
         with pytest.raises(RuntimeError, match='Compensation grade of ICA'):
             ica.get_sources(inst)
 
+    # The CTF data are uniquely well suited to testing the ICA.find_bads_
+    # methods, so do that while we're here.
+
+    raw = read_raw_ctf(ctf_fname, preload=True)
+    # derive reference ICA components and append them to raw
+    icarf = ICA(n_components=2, random_state=0, max_iter=2, allow_ref_meg=True)
+    with pytest.warns(UserWarning, match='did not converge'):
+        icarf.fit(raw.copy().pick_types(meg=False, ref_meg=True))
+    icacomps = icarf.get_sources(raw)
+    # rename components so they are auto-detected by find_bads_ref
+    for c in icacomps.ch_names:
+        icacomps.rename_channels({c: 'REF_' + c})
```

Comment:
`icacomps.rename_channels({c: 'REF_' + c for c in icacomps.ch_names})`

would be nicer and faster.